service/tasks/driver.py

```diff
@@ -623,6 +623,8 @@ def get_chain_from_build(
     """
     wait_active_task = wait_for_instance.s(
         instance.id, driverCls, provider, identity, "active")
+    add_security_group = add_security_group_task.si(driverCls, provider, core_identity, instance.id)
+    wait_active_task.link(add_security_group)
```

Comment:
Good find! Thanks! I have updated a new patch for this.

And for the `security_group_init`, we still need to have it since we need to create a security group for the user during the first launch. And that's something we could do before launching the instance. And assigning security group, which is handled by celery can only do after launching the instance. HTH. 👍